Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -996,6 +399,200 @@ bool CGImageIsMask(CGImageRef image) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatBmp, bytes, length);
+}
+
+CGImageRef _CGImageLoadTIFF(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatTiff, bytes, length);
+}
+
+CGImageRef _CGImageLoadPNG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatPng, bytes, length);
+}
+
+CGImageRef _CGImageLoadJPEG(void* bytes, int length) {
+    return _CGImageLoadImageWithWICDecoder(GUID_ContainerFormatJpeg, bytes, length);
+}
+
+CGImageRef _CGImageLoadImageWithWICDecoder(REFGUID decoderCls, void* bytes, int length) {
+    ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+
+    ComPtr<IWICBitmapDecoder> pDecoder;
+
+    if (decoderCls != GUID_NULL) {
+        ComPtr<IStream> spStream;
+        RETURN_NULL_IF_FAILED(::CreateStreamOnHGlobal(NULL, FALSE, &spStream));
+
+        ULONG written = 0;
+        HRESULT hr = spStream->Write(bytes, length, &written);
+        if (FAILED(hr) || (written != length)) {
+            TraceError(TAG, L"IStream::Write failed hr=%x len=%d written=%d", hr, length, written);
+            return NULL;
+        }
+
+        // seek to start
+        LARGE_INTEGER offset;
+        offset.QuadPart = 0;
+        RETURN_NULL_IF_FAILED(spStream->Seek(offset, STREAM_SEEK_SET, NULL));
+
+        RETURN_NULL_IF_FAILED(imageFactory->CreateDecoder(decoderCls, NULL, &pDecoder));
+
+        hr = pDecoder->Initialize(spStream.Get(), WICDecodeMetadataCacheOnLoad);
+        RETURN_NULL_IF_FAILED(hr);
+
+    } else {
+        ComPtr<IWICStream> spStream;
+        RETURN_NULL_IF_FAILED(imageFactory->CreateStream(&spStream));
+
+        RETURN_NULL_IF_FAILED(spStream->InitializeFromMemory(static_cast<unsigned char*>(bytes), length));
+        RETURN_NULL_IF_FAILED(imageFactory->CreateDecoderFromStream(spStream.Get(), nullptr, WICDecodeMetadataCacheOnDemand, &pDecoder));
+    }
+
+    ComPtr<IWICBitmapFrameDecode> bitMapFrameDecoder;
+    RETURN_NULL_IF_FAILED(pDecoder->GetFrame(0, &bitMapFrameDecoder));
+
+    CGImageRef imageRef = __CGImage::CreateInstance();
+    imageRef->ImageSource().Attach(bitMapFrameDecoder.Detach());
+    return imageRef;
+}
+
+NSData* _CGImagePNGRepresentation(CGImageRef image) {
+    return _CGImageRepresentation(image, GUID_WICPixelFormat32bppBGRA);
+}
+
+NSData* _CGImageJPEGRepresentation(CGImageRef image) {
+    return _CGImageRepresentation(image, GUID_WICPixelFormat32bppCMYK);
+}
+
+NSData* _CGImageRepresentation(CGImageRef image, REFGUID guid) {
+    RETURN_NULL_IF(!image);
+
+    ComPtr<IWICImagingFactory> imageFactory = _GetWICFactory();
+    ComPtr<IWICFormatConverter> imageFormatConverter;
+
+    RETURN_NULL_IF_FAILED(imageFactory->CreateFormatConverter(&imageFormatConverter));
+
+    RETURN_NULL_IF_FAILED(imageFormatConverter->Initialize(
+        image->ImageSource().Get(), guid, WICBitmapDitherTypeNone, nullptr, 0.f, WICBitmapPaletteTypeCustom));
+
+    size_t height, width;
```

Comment:
>size_t height, width; [](start = 4, length = 21)

super nit: prefer us to not do compound initialization as weird things can happen when you do this pointers, etc. Saving you time when coding it the first time is less important than saving maintainers time when reading and debugging issues.